### PR TITLE
Add support for mbedtls-3.x

### PR DESCRIPTION
**Pull request by gstrauss**
Add support for mbedtls-3.x

### CODE REVIEW

**Comment by gstrauss** [comment]
BTW, if you prefer to keep the patch smaller, the umurmur `urandom_bytes()` can be used instead of the mbedtls routines to get random bytes.

File: src/ssli_mbedtls.c [hunk 1]
```diff
@@ -37,9 +37,13 @@
...
-#include <mbedtls/config.h>
 #include <mbedtls/version.h>
-#include <mbedtls/havege.h>
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#include <mbedtls/psa_util.h>
+#else
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/entropy.h>
+#endif
 #include <mbedtls/certs.h>
```
Comment (Ansuel) [inline src/ssli_mbedtls.c:47]:
```suggestion
#if MBEDTLS_VERSION_MAJOR < 3
#include <mbedtls/certs.h>
#endif
```

This got dropped with mbedtls3

Comment (gstrauss) [reply src/ssli_mbedtls.c:48]: Yes, thank.  I incorporated your patch.

**Comment by Neustradamus** [comment]
Thanks to @gstrauss for the PR and @troxor for merging!

@troxor: A new release build can be done?

**Pull request merged**